gc/gc-base/src/test/java/org/projectnessie/gc/roundtrip/TestMarkAndSweep.java

```diff
@@ -0,0 +1,320 @@
+          + numExpired
+          + '}';
+    }
+
+    String numToHash(long l) {
+      return String.format("%016x", l);
+    }
+
+    String numToContentId(long l) {
+      return cidPrefix + l;
+    }
+
+    URI numToBaseLocation(long l) {
+      return basePath.resolve(l + "/");
+    }
+
+    long hashToNum(Reference ref) {
+      return Long.parseLong(ref.getHash(), 16);
+    }
+
+    long contentIdToNum(String cid) {
+      return Long.parseLong(cid.substring(cidPrefix.length()));
+    }
+
+    long baseLocationToNum(URI path) {
+      String p = path.getPath();
+      int l = p.lastIndexOf('/');
+      int l1 = p.lastIndexOf('/', l - 1);
+      return Long.parseLong(p.substring(l1 + 1, l));
+    }
+
+    Content numToContent(long l) {
+      return IcebergTable.of(
+          numToBaseLocation(l).resolve("metadata-" + l).toString(), l, 1, 2, 3, numToContentId(l));
+    }
+
+    ContentKey numToContentKey(long l) {
+      return ContentKey.of("content", Long.toString(l));
+    }
+
+    LogEntry numToLogEntry(long l) {
+      return LogEntry.builder()
+          .commitMeta(
+              CommitMeta.builder()
+                  .commitTime(now.minus(numCommits - l, ChronoUnit.SECONDS))
+                  .hash(numToHash(l))
+                  .message("hello")
+                  .build())
+          .parentCommitHash(numToHash(l - 1))
+          .addOperations(Put.of(numToContentKey(l), numToContent(l)))
+          .build();
+    }
+
+    long liveContentsCount() {
+      return numCommits + numKeysAtCutOff - numExpired;
+    }
+  }
+
+  @ParameterizedTest
+  @MethodSource("markAndSweep")
+  public void markAndSweep(MarkAndSweep markAndSweep) throws Exception {
+
+    InMemoryPersistenceSpi storage = new InMemoryPersistenceSpi();
+    LiveContentSetsRepository repository =
+        LiveContentSetsRepository.builder().persistenceSpi(storage).build();
+    IdentifyLiveContents identify =
+        IdentifyLiveContents.builder()
+            .contentTypeFilter(
+                new ContentTypeFilter() {
+                  @Override
+                  public boolean test(Content.Type type) {
+                    return ICEBERG_TABLE == type;
+                  }
+
+                  @Override
+                  public Set<Content.Type> validTypes() {
+                    return singleton(ICEBERG_TABLE);
+                  }
+                })
+            .cutOffPolicySupplier(r -> atTimestamp(markAndSweep.cutOff))
+            .contentToReference(
+                (content, commitId, key) ->
+                    ContentReference.icebergTable(
+                        content.getId(),
+                        commitId,
+                        key,
+                        ((IcebergTable) content).getMetadataLocation(),
+                        ((IcebergTable) content).getSnapshotId()))
+            .liveContentSetsRepository(repository)
+            .repositoryConnector(
+                new RepositoryConnector() {
```

Comment:
why are we not using the `NessieRepositoryConnector` for this end to end test?

Reply:
It's a unit test?